ext/opcache/ZendAccelerator.c

```diff
@@ -2003,11 +2003,14 @@ zend_op_array *persistent_compile_file(zend_file_handle *file_handle, int type)
 			return accelerator_orig_compile_file(file_handle, type);
 		}
 
+		HANDLE_UNBLOCK_INTERRUPTIONS();
```

Comment:
I'm not protecting SHM here because opcache_compile_file may increment `ZCSG(blacklist_misses)`. Maybe it is better to protect here and unprotect for that increment? It seems like there is potential for file compilation to corrupt SHM and currently we will miss such cases.

Reply:
I went ahead and did this now.